### PR TITLE
Split up VoxelLocation into immutable BaseLocation and mutable VoxelLocation

### PR DESCRIPTION
**Pull request by Lennart99**

### CODE REVIEW

File: VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/location/VoxelLocation.java [hunk 1]
```diff
@@ -115,81 +55,17 @@ public void add(VoxelVector direction) {
         this.addZ(direction.getZ());
     }
 
-    public IWorld getWorld() {
-        return this.world;
+    public BaseLocation makeImmutableLocation() {
```
Comment (KevinDaGame) [inline VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/location/VoxelLocation.java:58]:
```suggestion
    public BaseLocation makeImmutable() {
```

File: VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/location/BaseLocation.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    @Override
+    public BaseLocation clone() {
+        try {
+            return (BaseLocation) super.clone();
+        } catch (CloneNotSupportedException e) {
+            throw new Error(e);
+        }
+    }
+    
+    public VoxelLocation makeMutableLocation() {
```
Comment (KevinDaGame) [inline VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/location/BaseLocation.java:89]:
```suggestion
    public VoxelLocation makeMutable() {
```

**Review by KevinDaGame — APPROVED** [review]
LGTM! Just a small suggestion

File: VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/location/BaseLocation.java (hunk 2 above)
Comment (Lennart99) [reply VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/location/BaseLocation.java]: https://github.com/KevinDaGame/VoxelSniper-Reimagined/pull/93/commits/85b0a3490b0ac6720858f00e4d58e236bab298ce

File: VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/location/VoxelLocation.java (hunk 1 above)
Comment (Lennart99) [reply VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/location/VoxelLocation.java]: https://github.com/KevinDaGame/VoxelSniper-Reimagined/pull/93/commits/85b0a3490b0ac6720858f00e4d58e236bab298ce

**Pull request merged**